ivy_tests/test_ivy/helpers.py

```diff
@@ -1901,3 +1901,53 @@ def num_positional_args(draw, *, fn_name: str = None):
     return draw(
         integers(min_value=num_positional_only, max_value=(total - num_keyword_only))
     )
+
+
+def bool_val_flags(cl_arg: Union[bool, None]):
+    if cl_arg is not None:
+        return st.booleans().filter(lambda x: x == cl_arg)
+    return array_bools(num_arrays=1)
+
+
+def handle_cmd_line_args(test_fn):
+    # first four arguments are all fixtures
+    def new_fn(get_command_line_flags, fw, device, call, *args, **kwargs):
+        # inspecting for keyword arguments in test function
+        for param in inspect.signature(test_fn).parameters.values():
+            if param.kind == param.KEYWORD_ONLY:
+                if param.name == "data":
+                    data = kwargs["data"]
+                if param.name == "as_variable":
```

Comment:
from here below should all be `elif`, this will speed things up considerably (currently, we keep checking everything even after we have found a match.)